File: design-patterns/ts/visitor/ui-element/text-element.ts
// text-element.ts

import UIElement from "./ui-element";
import Visitor from "./visitor";

class TextElement implements UIElement {
    constructor(private text: string) {

    }

    getText(): string {
        return this.text
    }

    appendElement(visitor: Visitor): void {
        visitor.appendContent(this);
    }
}

export default TextElement;

File: design-patterns/go/visitor/ui_element/text_element.go
// text_element.go

package main

type TextElement struct {
	text string
}

func NewTextElement(text string) (textElement *TextElement) {
	textElement = &TextElement{}
	textElement.text = text
	return
}

func (textElement *TextElement) GetText() string {
	return textElement.text
}

func (textElement *TextElement) AppendElement(visitor Visitor) {
	visitor.AppendContent(textElement)
}
